tiledb/CMakeLists.txt

```diff
@@ -561,6 +624,76 @@ if (CURL_APPEND_LIBS)
     ${CURL_APPEND_LIBS})
 endif()
 
+############################################################
+# capnproto file generation
+############################################################
+
+if(TILEDB_SERIALIZATION)
+  if(TILEDB_CAPNP_EP_BUILT)
+    set(TILEDB_CAPNPROTOEXEPATH "${TILEDB_CAPNPEXEC_PATH}")
+  else()
+    find_package(CapnProto 0.6.1 #*not* EXACT since windows/*n*x using 0.8.0, 0.6.1 respectively
+      PATHS  ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH}
+    )
+    set(TILEDB_CAPNPROTOEXEPATH "${CAPNP_EXECUTABLE}")
+  endif()
+
+  #force (re-)gen to run each time so can alternate msvc/wsl builds without removing *-capnp.<h,c++>
+  if(WIN32)
+    add_custom_target(tiledb_genned_capnpfiles #ALL
+      COMMAND
+        powershell "../scripts/capnpupdate.ps1 "
+      WORKING_DIRECTORY
+        ${TILEDB_EP_BASE}/..
+    )
+  else()
+    add_custom_target(tiledb_genned_capnpfiles #ALL
+      COMMAND
+        bash -c "../scripts/capnpupdate.sh "
+      WORKING_DIRECTORY
+        ${TILEDB_EP_BASE}/..
+    )
+  endif()
+  add_dependencies(TILEDB_CORE_OBJECTS tiledb_genned_capnpfiles)
+
+  #make files 'known' via add_custom_command() so ("tiledb-rest.capnp.c++") can be added to TILEDB_CORE_OBJECTS (further below)
+  if(WIN32)
+    add_custom_command(
+      OUTPUT #BYPRODUCTS
+        ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.c++
+        ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.h
+      COMMAND
+        powershell "../scripts/capnpupdate.ps1 "
+      WORKING_DIRECTORY
+        ${TILEDB_EP_BASE}/..
+
+      COMMENT "(re-)generate tiledb-rest.capnp.<h,c++>"
+    )
+  else()
+    add_custom_command(
+      OUTPUT #BYPRODUCTS
+        ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.c++
+        ${TILEDB_CORE_INCLUDE_DIR}/tiledb/sm/serialization/tiledb-rest.capnp.h
+      COMMAND
+        bash -c "../scripts/capnpupdate.sh "
+      WORKING_DIRECTORY
+        ${TILEDB_EP_BASE}/..
+      COMMENT "(re-)generate tiledb-rest.capnp.<h,c++>"
+    )
+  endif()
+
+  # somethingnotgennedforcingcapntoalwaysrun is not actually generated so set it to symbolic
+  #TBD: but this is problematic for multiple build attempts against same
+  #cmake config as re-patching does not work.
+  set_source_files_properties(somethingnotgennedforcingcapntoalwaysrun PROPERTIES GENERATED true)
+
```

Comment:
What is this doing? Could we use a less verbose variable name?

Reply:
removed, not currently useful with gen'd files going to respective target (*nx/windows) build directories